ark/internal/controller/query_controller.go

```diff
@@ -34,10 +33,19 @@ type targetResult struct {
 	target   arkv1alpha1.QueryTarget
 }
 
+// QueryReconciler reconciles a Query object with telemetry abstraction.
```

Comment:
this looks useful but we might need 2-3 bullets into the telemetry docs instead, any thoughts on where is best to put this?